solidity/contracts/middleware/InterchainAccountMirror.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+// ============ Internal Imports ============
+import {TypeCasts} from "../libs/TypeCasts.sol";
+import {InterchainAccountRouter} from "./InterchainAccountRouter.sol";
+import {CallLib} from "./libs/Call.sol";
+import {InterchainAccountMirrorCalldata} from "./libs/InterchainAccountMirrorCalldata.sol";
+
+contract InterchainAccountMirror {
```

Comment:
Obiouvlsy im not actually a naming czar, but ngl this sounds p unintuitive to me. I think the word proxy would make more sense than mirror

Reply:
proxy is already used for the other end and I didnt want to conflate the two

Reply:
I personally spent a bit of time thinking about this and arrived at mirror

Reply:
ingress and egress proxy?

Reply:
given this is not intended for external use atm the name does not seem important